### ISSUE TITLE
Could you please add an editor?

### ISSUE THREAD

**Issue by SoNewBee** [Urho-Net/Urho.Net#19]
This is a brilliant project.
I am a noob and I wish you could add an editor and make some showcase.
Appreciated!

**Comment by elix22** [comment]
I am Working on an (Unity-light like) cross platform Net 6 Editor , It's quite a big project , don't have an estimate of release date (probably preview will be available in Q3/Q4 2022).

Only GL/GLES backend is supported (on iOS also Metal)  ,  DX backend is not supported , I don't see any reason to enable it

You can build your own binary (with DX rendering backend enabled) , I didn't tried and I don't plan to support it officially.
The Link for the build script for Windows https://github.com/elix22/Urho3D/blob/dotnet/script/build_install_urhonet_windows.ps1
Which on it's turn invokes the script (you have to modify it)
https://github.com/elix22/Urho3D/blob/dotnet/script/cmake_vs2019_dotnet_dll.bat

Remove -D URHO3D_OPENGL=1 
Add -D URHO3D_D3D11=1
It should build and install a binary supporting DX11 instead of GL (on Windows)

**Comment by SoNewBee** [comment]
> I am Working on an (Unity-light like) cross platform Net 6 Editor , It's quite a big project , don't have an estimate of release date (probably preview will be available in Q3/Q4 2022).
> 
> Only GL/GLES backend is supported (on iOS also Metal) , DX backend is not supported , I don't see any reason to enable it
> 
> You can build your own binary (with DX rendering backend enabled) , I didn't tried and I don't plan to support it officially. The Link for the build script for Windows https://github.com/elix22/Urho3D/blob/dotnet/script/build_install_urhonet_windows.ps1 Which on it's turn invokes the script (you have to modify it) https://github.com/elix22/Urho3D/blob/dotnet/script/cmake_vs2019_dotnet_dll.bat
> 
> Remove -D URHO3D_OPENGL=1 Add -D URHO3D_D3D11=1 It should build and install a binary supporting DX11 instead of GL (on Windows)

Thanks for your reply.
That is very excited.I can't wait to see that editor!GOOD JOB!Finally I can get rid of c++!
Maybe you can remove all files that related to DirectX since this project will not do anything with dx(something like hlsl folder and etc.?).
And...maybe you can make the project more succinct.Because the temporary version is too big(about 941MB,No offence,Godot with its editor is about 50MB or smaller).
And...Will you add vulkan support in the future?

**Comment by elix22** [comment]
Yep there are some redundant files in the Assets folder , I might make some cleanup in the future.
Godot is definitely more than 50 MB , Godot Export templates are 534 MB in size (compressed form) 
I don't plan to reduce the size , actually it will increase relatively for each new/additional platform support
No Vulkan support in the future , however Graphics enhancements are planned .
I implement only whatever is needed/helpful for my own game projects , Vulkan is not one of those things .

**Comment by SoNewBee** [comment]
> Yep there are some redundant files in the Assets folder , I might make some cleanup in the future. Godot is definitely more than 50 MB , Godot Export templates are 534 MB in size (compressed form) I don't plan to reduce the size , actually it will increase relatively for each new/additional platform support No Vulkan support in the future , however Graphics enhancements are planned . I implement only whatever is needed/helpful for my own game projects , Vulkan is not one of those things .

OK...So this project won't support multi-thread rendering?Can I understand it as this?

**Comment by elix22** [comment]
Multithreaded rendering is not the main bottleneck for generating beautiful scenery .
Currently I can't elaborate on the future graphics enhancements .

**Comment by SoNewBee** [comment]
> Multithreaded rendering is not the main bottleneck for generating beautiful scenery . Currently I can't elaborate on the future graphics enhancements .

So...Can I understand it as this project will not support multi-threaded rendering?
